### DISCUSSION TITLE
Pre-event before installation/update takes place.

### DISCUSSION THREAD

**Discussion by Minisaints** [ravibpatel/AutoUpdater.NET#675] [General]
Hello,

I'm looking to update a windows service (.exe) along with the automatic update as part of the update ZIP. However, I have to stop the service, replace the executable and start it.

Where can I trigger stopping and starting the windows service if an update is found? I don't see any pre-events such as "beforeUpdateEvent" etc.

Thanks.

**Comment by Minisaints** [comment]
Found a way around it. I parse the manifest, re-check the versions and down the service. Not ideal as I have to restart the service elsewhere but it's a solution for now.

**Comment by ravibpatel** [comment]
You can just handle the update flow manually as shown [here](https://github.com/ravibpatel/AutoUpdater.NET?tab=readme-ov-file#handling-updates-manually). You can stop the service if update is found and then start the Download. I think it will be better idea to create a separate exe to update the servcie that can handle the stopping and starting the service.
